### DISCUSSION TITLE
Can we apply mid force?

### DISCUSSION THREAD

**Discussion by vkapre** [GazzolaLab/PyElastica#273] [Q&A] [answered]
Hello, 

I am new to the community. I would like to model the deformation of a slender fiber. From the documentation, it looks like I can only apply end forces or uniform forces. Can I apply a force anywhere along the length of the fiber? If not, would I have to split the fiber into two rods and apply the force as EndForce? I don't see any connection that would capture a continuous smooth fiber. I apologize if the question is basic or silly. 

Thank you.

**Comment by ilianasiriziba** [comment, marked as answer]
Hi @vkapre ,

Yes, you can do that on a single rod. You must define a custom force class (based on NoForces class) like the ones in the external_forces.py file. Then in the apply_forces function you have access to system.external_forces where you can apply forces on each node you want. 

Let me know if it helps

**Reply by vkapre** [reply]
Thank you so much for your suggestion. I added a custom force class called 'Midpointforces' in the external_forces.py code. Do I have to make any other changes? When I try to implement it in my model, I get the error "AttributeError: module 'elastica' has no attribute 'MidpointForces'. Did you mean: 'EndpointForces'?". Where is the apply_forces function? Thanks again.

**Reply by ilianasiriziba** [reply]
If you look at the force classes in the external_forces.py you will find the apply_forces and how the force is applied on the nodes in different classes. Also, are you sure that name of the class you created is the same as the one you called? Because it seems you created 'Midpointforces' class but you called 'MidpointForces' which are not same.
